lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -784,6 +784,11 @@ def signature_of(method)
           nil
         end
 
+        sig { params(constant: Module).returns(String) }
+        def type_of(constant)
+          constant.to_s.gsub(/AttachedClass/, "T.attached_class")
```

Comment:
Ugh, would this not make something like `T::Array[ModelAttachedClass]` into `T::Array[ModelT.attached_class]`?

Reply:
`/\bAttachedClass\b/` is what we want, I think:
```ruby
[12] pry(main)> "ModelAttachedClass".gsub(/AttachedClass/, "T.attached_class")
=> "ModelT.attached_class"
[13] pry(main)> "ModelAttachedClass".gsub(/\bAttachedClass\b/, "T.attached_class")
=> "ModelAttachedClass"
[14] pry(main)> "[T::Array[ModelAttachedClass], [AttachedClass, String]]".gsub(/\bAttachedClass\b/, "T.attached_class")
=> "[T::Array[ModelAttachedClass], [T.attached_class, String]]"
```

Reply:
Good catch, I also added your test 👍